### PR TITLE
Feat/my profile

### PR DESCRIPTION
**Pull request by MoinMarcell**

### CODE REVIEW

**Review by MarcusSchweizer91 — APPROVED** [review]

**Review by MarcusSchweizer91 — APPROVED** [review]

File: backend/src/main/java/com/github/moinmarcell/backend/service/FliprUserService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    public Optional<FliprUserDTO> getFliprUserByUsername(String username){
+        Optional<FliprUser> user = mongoUserRepo.findByUsername(username);
+        return user.map(fliprUser -> new FliprUserDTO(fliprUser.fliprID(), fliprUser.username(), fliprUser.email(), fliprUser.fliprList()));
+    }
+
+    public Optional<FliprUserDTO> getFliprUserByFliprID(String fliprID){
+        Optional<FliprUser> user = mongoUserRepo.findFliprUserByFliprID(fliprID);
+        return user.map(fliprUser -> new FliprUserDTO(fliprUser.fliprID(), fliprUser.username(), fliprUser.email(), fliprUser.fliprList()));
+    }
```
Comment (Droggelbecher92) [inline backend/src/main/java/com/github/moinmarcell/backend/service/FliprUserService.java:31]: same code 3 times, maybe time private helper method to create DTO's ;)

File: backend/src/main/java/com/github/moinmarcell/backend/controller/FliprUserController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        this.fliprUserService = fliprUserService;
+    }
+
+    @GetMapping
+    public List<FliprUserDTO> getAllFliprUsers(){
+        return fliprUserService.getAllFliprUsers();
+    }
+
+    @GetMapping("/user")
+    public Optional<FliprUserDTO> getFliprUserByUsername(@RequestParam(required = false) String username, @RequestParam(required = false) String fliprID){
```
Comment (Droggelbecher92) [inline backend/src/main/java/com/github/moinmarcell/backend/controller/FliprUserController.java:29]: Wouldn't return an Optional in your Controller, it's bad practice. Better use ResponseEntity as return type.

File: frontend/src/components/Navigation/Navigation.tsx [hunk 3]
```diff
@@ -21,10 +23,19 @@ const Navigation = (props: NavigationProps) => {
...
+                </div>
+            </div>
+
+            <div className={"row"}>
+                <div className={"col text-white text-end"}>
+                    {isAuthenticated ? <Link to={"/"} className={"text-white text-decoration-none"} onClick={props.logout}><i className="fa-solid fa-right-from-bracket"></i></Link> : ''}
                 </div>
                 <div className={"col text-start d-none d-md-block"}>
-                    {props.username === 'anonymousUser' ? <Link to={"/login"} className={"text-white text-decoration-none"}>Login</Link> : <span className={"text-white text-decoration-none curso"} onClick={props.logout}>{props.username}</span>}
+                    {isAuthenticated ? <Link to={"/"} className={"text-white text-decoration-none curso"} onClick={props.logout}>Logout</Link> : ''}
```
Comment (Droggelbecher92) [inline frontend/src/components/Navigation/Navigation.tsx:38]: If more buttons will be added, I would think about a button component to make it cleaner. There are already 4 mostly identical divs.

**Review by Droggelbecher92 — APPROVED** [review]
Love the amount of tests! lgtm 👌

**Pull request merged**